src/tfr_control/arduino/encoder/encoder.ino

```diff
@@ -1,41 +1,121 @@
 #include <Encoder.h>
+#include <Wire.h>
+#include <Adafruit_ADS1015.h>
 #include <ros.h>
-#include <tfr_msgs/EncoderReading.h>
+#include <tfr_msgs/ArduinoReading.h>
 #include <quadrature.h>
 
 ros::NodeHandle nh;
 
 //encoder level constants
-const double GEARBOX_CPR = 4096; //pulse per revolution
-//TODO note this rev/m is acually 0.87, but I had to step it up to make testing work
+const double CPR = 4096; //pulse per revolution
+//TODO note this rev/m is acually 0.876, but I had to step it up to make testing work
 const double GEARBOX_RPM = 60; //revolutions per meter
+//TODO note this rev/m is acually 0.208, but I had to step it up to make testing work
+const double TURNTABLE_RPM = 60;
 
 //pin constants
 const int GEARBOX_LEFT_A = 2;
 const int GEARBOX_LEFT_B = 3;
 
+
+
+
+struct Potentiometer
+{
+   const static float ADC_RANGE=0.0001875;
+   float max_voltage; //the max voltage for the potentiometer
+   float min_angle;
+   float max_angle;
+   float getPosition(int16_t val)
+   {
+       return min_angle + (max_angle - min_angle) * val* ADC_RANGE/max_voltage; 
+   }
+};
+
+enum Potentiometers
+{
+    //used for array indexes
+    ARM_LOWER_LEFT,
+    ARM_LOWER_RIGHT,
+    ARM_UPPER,
+    ARM_SCOOP,
+    BIN_LEFT,
+    BIN_RIGHT
+};
+
+Potentiometer pots []
+{
+  {4.88,  0.209, 1.693},      //ARM_LOWER_LEFT
+  {4.88,  0.209, 1.693},      //ARM_LOWER_RIGHT
+  {4.88,  0.000, 1.868},      //ARM_UPPER
+  {4.88, -0.873, 1.868},      //ARM_SCOOP
+  {4.88,  0.000, 0.786},      //BIN_LEFT
+  {4.88,  0.000, 0.786}       //BIN_RIGHT
+};
+
+
 //encoders
-Quadrature gearbox_left(GEARBOX_CPR, GEARBOX_LEFT_A, GEARBOX_LEFT_B);
-tfr_msgs::EncoderReading reading;
+VelocityQuadrature gearbox_left(CPR, GEARBOX_LEFT_A, GEARBOX_LEFT_B);
+//TODO Quadrature gearbox_right(CPR, GEARBOX_LEFT_A, GEARBOX_LEFT_B);
+//TODO Quadrature turntable(CPR, GEARBOX_LEFT_A, GEARBOX_LEFT_B);
```

Comment:
This is just because we don't have things to plug in yet?

Reply:
Yup!

Reply:
In theory when we plug in the the hardware devices, and wire them we should be able to uncomment the line, recompile and be good to go. I have tested everything I can on a comprable device.